lib/mongoid/persistence_context.rb

```diff
@@ -232,7 +231,39 @@ def clear(object, cluster = nil, original_context = nil)
           end
         end
       ensure
-        Thread.current["[mongoid][#{object.object_id}]:context"] = original_context
+        store_context(object, original_context)
+      end
+
+      private
+
+      # Get the persistence context for a given object from the thread local
+      #   storage.
+      #
+      # @param [ Object ] object Object to get the persistance context for.
+      #
+      # @return [ Mongoid::PersistenceContext | nil ] The persistence context
+      #   for the object if previously stored, otherwise nil.
+      #
+      # @api private
+      def get_context(object)
+        Thread.current["mongoid:persistence_context"] ||= {}
```

Comment:
Let's stick with `[mongoid]` prefix which is used everywhere except for https://jira.mongodb.org/browse/MONGOID-5441.

Reply:
Done